pkg/node/addressing/addresstype.go

```diff
@@ -16,3 +21,47 @@ const (
 	NodeInternalDNS      AddressType = "InternalDNS"
 	NodeCiliumInternalIP AddressType = "CiliumInternalIP"
 )
+
+type Address interface {
+	AddrType() AddressType
+	ToString() string
+}
+
+// GetIP returns one of the provided IP addresses available with the following priority:
```

Comment:
nit:

```suggestion
//  ExtractNodeIP returns one of the provided IP addresses available with the following priority:
```

Reply:
Thanks for catching this mistake @gandro. Commit 4a735bb resolves your comment.